### PR TITLE
Show error message to SupportUser if they try to revert a withdrawal when it creates a validation error

### PR DESCRIPTION
**Pull request by inulty-dfe**
## Context

When a candidate has applied to a course, then that application is withdrawn, then they add the course again to their applcaition form and then someone tries to revert the original application, we get an error. This causes a 500 exception but now we present an error message instead.

## Changes proposed in this pull request

- Show error message instead of raising exception.
- Inherit from SimpleDelegator so all errors and validate method are called on the delegatee.
- Maybe the error message needs to change based on whether it's presented to the Candidate or the Support user.
- No need to validate presence and format of zendesk URL, if the URL is blank, the format will be enough.

## Guidance to review

Test it out in the REview app https://apply-review-9587.test.teacherservices.cloud/support/applications/30

[revert_withdrawal.webm](https://github.com/user-attachments/assets/6c4c29a4-7353-4fcb-a186-d95cd435f627)

## Link to Trello card

[Trello Ticket](https://trello.com/c/UaEJEYjm/2082-show-error-message-to-provider-user-if-they-try-to-revert-a-withdrawal-when-it-creates-a-validation-error)

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by inulty-dfe** [comment]
[Show error message to Provider User if they try to revert a withdrawal when it creates a validation error](https://trello.com/c/UaEJEYjm/2082-show-error-message-to-provider-user-if-they-try-to-revert-a-withdrawal-when-it-creates-a-validation-error)

File: app/forms/support_interface/application_forms/revert_withdrawal_form.rb [hunk 1]
```diff
@@ -3,20 +3,33 @@ module ApplicationForms
...
-        return false unless valid?
+        validate
 
-        SupportInterface::RevertWithdrawal.new(
+        return false if errors.any?
+
+        revert_withdrawal_service.save
+      end
+
+      def revert_withdrawal_service
```
Comment (CatalinVoineag) [inline app/forms/support_interface/application_forms/revert_withdrawal_form.rb:22]: Should this be a private method if you're only using it in this class?

**Review by elceebee — APPROVED** [review]

**Review by CatalinVoineag — APPROVED** [review]
Looks good, one comment, not a blocker

File: app/forms/support_interface/application_forms/revert_withdrawal_form.rb (hunk 1 above)
Comment (avinhurry) [reply app/forms/support_interface/application_forms/revert_withdrawal_form.rb:28]: Makes sense, updated here.

https://github.com/DFE-Digital/apply-for-teacher-training/pull/9587/commits/4dc422eeb77d5c4c9f61f436761b942c76af5131

**Pull request merged**